Packs/AWS-IAMIdentityCenter/Integrations/AWSIAMIdentityCenter/AWSIAMIdentityCenter.py

```diff
@@ -2,244 +2,427 @@
+        outputs=ec
     )
-    ec = {'AWS.IAMIdentityCenter.Users': response}
-    human_readable = tableToMarkdown('AWS IAM Identity Center Users', response)
-    return_outputs(human_readable, ec)
+    return_results(result)
 
 
-def get_user(args, client):  # pragma: no cover
-    data = []
-    userName = demisto.getArg('userName')
-    response = client.list_users(
-        IdentityStoreId=f'{IDENTITYSTOREID}',
-        Filters=[
-            {
-                'AttributePath': 'UserName',
+def get_userId_by_username(args, client, IdentityStoreId):
+    userName = args.get('userName')
+    response_id = client.get_user_id(
+        IdentityStoreId=f'{IdentityStoreId}',
+        AlternateIdentifier={
+            'UniqueAttribute': {
+                'AttributePath': "userName",
                 'AttributeValue': f'{userName}'
-            },
-        ]
-    )
-    rep = json.dumps(response)
-    repJSON = json.loads(rep)
-    datas = repJSON.get('Users', [])
-    for da in datas:
-        for user in response['Users']:
-            user_details = {
-                'UserName': user['UserName'],
-                'UserId': user['UserId'],
-                'Email': user['Emails'][0]['Value'],
-                'DisplayName': user['DisplayName']
             }
-            userID = user['UserId']
-            data.append(user_details)
-    ec = {'AWS.IAM.IdentityCenter.Users': data}
-    human_readable = tableToMarkdown('AWS IAM Users', data, removeNull=True)
-    return_outputs(human_readable, ec)
-    return userID
+        }
+    )
+    return response_id
 
 
-def get_user_by_email(args, client):  # pragma: no cover
-    data = []
-    emailArg = demisto.getArg('emailAddress')
-    response = client.list_users(
-        IdentityStoreId=f'{IDENTITYSTOREID}',
+def delete_user(args, client, IdentityStoreId):
+    userId = get_userId_by_username(args, client, IdentityStoreId)['UserId']
+    response = client.delete_user(
+        IdentityStoreId=f'{IdentityStoreId}',
+        UserId=f'{userId}'
     )
-    rep = json.dumps(response)
-    repJSON = json.loads(rep)
-    datas = repJSON.get('Users', [])
-    for da in datas:
-        for user in response['Users']:
-            userEmail = user['Emails'][0]['Value']
-            if userEmail == emailArg:
-                user_details = {
-                    'UserName': user['UserName'],
-                    'UserId': user['UserId'],
-                    'Email': user['Emails'][0]['Value'],
-                    'DisplayName': user['DisplayName']
-                }
-                userID = user['UserId']
-                data.append(user_details)
-    ec = {'AWS.IAM.IdentityCenter.Users': data[0]}
-    human_readable = tableToMarkdown('AWS IAM Users ', data[0], removeNull=True)
-    return_outputs(human_readable, ec)
-    return userID
-
-
-def list_users(args, client):  # pragma: no cover
-    data = []
+    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
```

Comment:
What happens if the status code is not 200 you return None? an error result would be expected. 
```suggestion
    if response.get('ResponseMetadata', {}).get('HTTPStatusCode') == 200:
```